sql/src/test/java/io/crate/planner/operators/LogicalPlannerTest.java

```diff
@@ -200,6 +202,28 @@ public void testScoreColumnIsCollectedNotFetched() throws Exception {
                                 "Collect[doc.t1 | [_fetchid, _score] | All]\n"));
     }
 
+    @Test
+    public void testInWithSubqueryOrderImplicitlyAppliedOrdersMerged() {
+        LogicalPlan plan = plan("select x from t1 where x in (select x from t1 order by x desc)");
+        assertThat(plan.dependencies().entrySet().size(), is(1));
+        LogicalPlan subPlan = plan.dependencies().keySet().iterator().next();
+        assertThat(subPlan, isPlan("RootBoundary[x]\n" +
+                                   "OrderBy[x][false][false]\n" +
+                                   "Collect[doc.t1 | [x] | All]\n"));
+    }
+
+    @Test
+    public void testInWithSubqueryOrderWithExistingOrderByImplicitlyApplied() {
+        LogicalPlan plan = plan("select x from t1 where x in (select x from t1 order by 1 desc limit 10)");
+        assertThat(plan.dependencies().entrySet().size(), is(1));
+        LogicalPlan subPlan = plan.dependencies().keySet().iterator().next();
+        assertThat(subPlan, isPlan("RootBoundary[x]\n" +
+                                   "OrderBy[x][false][false]\n" +
+                                   "Limit[10;0]\n" +
+                                   "OrderBy[x][true][null]\n" +
```

Comment:
Why do we order twice here? Shouldn't this PR improve performance? :)

Reply:
Because we add the order that we need for the TermInSetQuery and then we let the tryOptimize do the optimization. In this case we have a limit, so the tryOptimize cannot pass the Order through it. Check discussion with @mfussenegger to add some more clever check if we should add the Ordering or not.

Reply:
I wouldn't do this optimization if it can lead to worse query performance.

Reply:
@mxm check the comment above